test/e2e/registration_service_test.go

```diff
@@ -429,6 +431,177 @@ func (s *registrationServiceTestSuite) TestSignupOK() {
 	assert.Equal(s.T(), ExpectedConsoleURL(s.T(), s.memberAwait, memberCluster), mp["consoleURL"])
 }
 
+func (s *registrationServiceTestSuite) TestPhoneVerification() {
+	// Create a token and identity to sign up with
+	identity0 := authsupport.NewIdentity()
+	emailValue := uuid.NewV4().String() + "@some.domain"
+	emailClaim0 := authsupport.WithEmailClaim(emailValue)
+	token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0)
+	require.NoError(s.T(), err)
+
+	// Call the signup endpoint
+	req, err := http.NewRequest("POST", s.route+"/api/v1/signup", nil)
+	require.NoError(s.T(), err)
+	req.Header.Set("Authorization", "Bearer "+token0)
+	req.Header.Set("content-type", "application/json")
+	resp, err := httpClient.Do(req)
+	require.NoError(s.T(), err)
+	defer close(s.T(), resp)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), body)
+	assert.Equal(s.T(), http.StatusAccepted, resp.StatusCode)
+
+	// Wait for the UserSignup to be created
+	userSignup, err := s.hostAwait.WaitForUserSignup(identity0.ID.String(), wait.UntilUserSignupHasConditions(PendingApproval()...))
+	require.NoError(s.T(), err)
+	emailAnnotation := userSignup.Annotations[v1alpha1.UserSignupUserEmailAnnotationKey]
+	assert.Equal(s.T(), emailValue, emailAnnotation)
+
+	// Call get signup endpoint with a valid token and make sure it's pending approval
+	req, err = http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+	require.NoError(s.T(), err)
+	req.Header.Set("Authorization", "Bearer "+token0)
+	req.Header.Set("content-type", "application/json")
+
+	resp, err = httpClient.Do(req)
+	require.NoError(s.T(), err)
+	defer close(s.T(), resp)
+
+	body, err = ioutil.ReadAll(resp.Body)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), body)
+
+	mp := make(map[string]interface{})
+	err = json.Unmarshal([]byte(body), &mp)
+	require.NoError(s.T(), err)
+
+	mpStatus, ok := mp["status"].(map[string]interface{})
+	assert.True(s.T(), ok)
+
+	assert.Equal(s.T(), "", mp["compliantUsername"])
+	assert.Equal(s.T(), identity0.Username, mp["username"])
+	require.IsType(s.T(), false, mpStatus["ready"])
+	assert.False(s.T(), mpStatus["ready"].(bool))
+	assert.Equal(s.T(), "PendingApproval", mpStatus["reason"])
+	require.True(s.T(), mpStatus["verificationRequired"].(bool))
+
+	// Approve usersignup.
+	userSignup.Spec.Approved = true
+	err = s.hostAwait.Client.Update(context.TODO(), userSignup)
+	require.NoError(s.T(), err)
```

Comment:
you know what is weird to me - that we firstly approve and then wait for the verification. Shouldn't that be vice versa? Approve only when it's verified?

Reply:
Yeah I guess that would be less ambiguous...

Reply:
That workflow actually seems a little weird to me, if I'm a user and I'm confronted with a phone verification requirement which I go to the effort to complete by providing my personal phone number, I can imagine being more than a little annoyed if after that I'm *still* required to wait for someone to manually approve me.  If I know that I need to wait for approval in the first place, then the phone verification step won't be a blocker for me to start using the product once someone has approved me.

Reply:
Phone verification is basically a prove that the user is not a bot. In many systems you can't even login without going through some verification. Including 2FA.
So, asking for phone verification (so the user is not a bot) before we even consider the user to be granted access to the system is IMO justified. We can double check with Parag though.

Reply:
I would definitely like to double check on this, I still feel uncomfortable about forcing a user to provide their phone number with the possibility that they still might potentially need to wait for an extended period of time to be approved.  What about situations where we have run out of cluster resources - what is the expected process here and what will the delay be for the user?  What if there are thousands of users in the queue waiting to be approved?

Reply:
Also in response to the comment about many existing systems requiring verification, I am struggling to think of a single example in which there might still be a required manual approval step after already verifying.

Reply:
We have manual approval step now for CRT cluster. I guess that we won't have it as soon as we go live. But it can happen that the clusters are full, then the user would need to wait.

Anyway, if I compare two types of the user experience:

First one that requires inputs from the user in two phases:
1. Sign up for the Dev Sandbox (that may require also creating a new user account at redhat.com)
2. wait until my account is approved (can be minutes, hours or days)
3. add additional information for phone verification (with the fact that the phone can be rejected since it was already used for some other user account)
4. Finally, enter Dev Sandbox

or provide all information at the beginning and then just wait until it is approved:
1. Sign up for the Dev Sendbox (that may require also creating a new user account at redhat.com) & verify the phone
2. Wait until my account is approved (can be minutes, hours or days)
3. Enter Dev Sandbox

To be honest, the second one seems to be more logical and fluent. I as a user give all necessary information about myself (including phone verification) at the beginning, then I just wait until I'm approved.

Reply:
+1 @MatousJobanek

Reply:
Actually looking at the code, this conversation is moot.... if verification is required then the `verificationRequired` property will be set to true regardless of whether the usersignup is approved or not, meaning that a new user should immediately have to perform phone verification after signing up anyway.  I've modified the test workflow to do the approval after the verification has been completed, in order to reflect this.

Reply:
Yes, you are right. :+1: 
I'm going to swap the verification & approval phases in my PR anyway: https://github.com/codeready-toolchain/host-operator/pull/288/files#diff-5b4b18a0fc2e0d708e4e6334cdb6bc69L285-L294 so I'm glad that we agreed on the flow